snippets/header-search.liquid

```diff
@@ -0,0 +1,97 @@
+{% comment %}
+  Renders a header search modal. Should be used with 'header.liquid'
+
+  Accepts:
+  - is_duplicate: {Boolean} Prevents duplicate ids for subsequent instances
```

Comment:
I'm not sure if my point was received, but this instance definitely doesn't make as strong of a case for that prop pattern as the other snippet I was matching so I decided to adjust.

> I wonder if we even need the id and for attributes if we could achieve the same thing by having the label as the parent of the input?

Well I definitely don't want to refactor every input in dawn to avoid a naming discussion regardless haha, but I imagine a major reason we chose the label for input pattern is so the label could access the :focus context of the input element in css (input:focus ~ label). Off hand I'm not aware of every detail considered in original decision, but I'm sure it wasn't made by accident or with ignorance to the label parent option.